tachyon/zk/plonk/permutation/grand_product_argument.h

```diff
@@ -55,7 +57,7 @@ class GrandProductArgument {
   }
 
  private:
-  FRIEND_TEST(LookupArgumentRunnerTest, ComputePermutationProduct);
+  friend class zk::LookupArgumentRunnerTest_ComputePermutationProduct_Test;
```

Comment:
What does this new random class do? There are other instances of private `FRIEND_TEST`s that aren't changed like this like in `selector_compressor.h?`

Reply:
This is what `FRIEND_TEST()` does. 

```c++
FRIEND_TEST(zk::LookupArgumentRunnerTest, ComputePermutationProduct);
// equivalent to
// friend class LookupArgumentRunnerTest_ComputePermutationProduct_Test;
```

I don't know the exact reason, but doing only above can't look up the symbol correctly. So we need a forward declaration above, then we end up with a question. If we do this ourselves, do we actually need `FRIEND_TEST()`? This question lead us to just flatten like above ourselves.